src/components/localtunneldebugger/localtunneldebugger.ts

```diff
@@ -0,0 +1,46 @@
+import * as vscode from 'vscode';
+
+import { LocalTunnelDebugProvider } from './localtunneldebugger.extension';
+import { getLocalTunnelDebugProvider } from '../config/config';
+
+export class LocalTunnelDebugger {
+    private readonly providers = Array.of<LocalTunnelDebugProvider>();
```

Comment:
💡 adding some thoughts, looks like cool feature. Is there any chances of this being `undefined`? or there is a default value fallback for this?

Reply:
Hey @Tatsinnit, would you mind clarifying this one this a little more? :) I was trying to follow the way cloudexplorer keeps track of its providers, but if I'm doing something unusual here please do let me know!

Reply:
I think the PR for the fort below covered some parts of this: 

* Just for other readers: https://github.com/Azure/vscode-kubernetes-tools/pull/851#discussion_r552996801